flake.nix

```diff
@@ -9,24 +9,27 @@
         nixpkgs = self;
       };
 
-      lib = import ./lib;
+      lib = (import ./lib).extend (final: prev: {
+        trivial = prev.trivial // {
+          versionSuffix = let
+            lastModified = final.substring 0 8 (toString
+              (self.lastModifiedDate or self.lastModified or "19700101"));
+            shortRev = self.shortRev or self.dirtyShortRev or "dirty";
+          in ".${lastModified}.${shortRev}";
+
+          revisionWithDefault = default: self.rev or self.dirtyRev or default;
+        };
+      });
 
       forAllSystems = lib.genAttrs lib.systems.flakeExposed;
     in
     {
       lib = lib.extend (final: prev: {
-
-        nixos = import ./nixos/lib { lib = final; };
+        nixos = import ./nixos/lib { lib = prev; };
```

Comment:
Why not `final`?

Reply:
AFAIK `nixos/lib` usually takes a plain Nixpkgs `lib`, not one extended with the `nixos` self-reference and `nixosSystem`, which is why I made this change. But maybe there's a reason it should see those when used through `nixpkgs.lib.nixos.*`?

Reply:
Another option would be to just use `//` here rather than another `.extend`? As far as I can tell they're just being added as exported convenience attributes rather than really extending the Nixpkgs `lib` in any meaningful way.

Reply:
Pushed a change to do that instead as it's hopefully clearer as to the intent and should have less overhead.